configuration/azure/appconfig/appconfig.go

```diff
@@ -295,21 +292,32 @@ func (r *ConfigurationStore) Subscribe(ctx context.Context, req *configuration.S
 }
 
 func (r *ConfigurationStore) doSubscribe(ctx context.Context, req *configuration.SubscribeRequest, handler configuration.UpdateHandler, sentinelKey string, id string) {
+	var etagVal *azcore.ETag
 	for {
-		// get sentinel key changes
-		_, err := r.Get(ctx, &configuration.GetRequest{
-			Keys:     []string{sentinelKey},
-			Metadata: req.Metadata,
-		})
+		// get sentinel key changes.
+		resp, err := r.getSettings(
+			ctx,
+			sentinelKey,
+			&azappconfig.GetSettingOptions{
+				Label:         r.getLabelFromMetadata(req.Metadata),
+				OnlyIfChanged: etagVal,
+			},
+		)
 		if err != nil {
-			r.logger.Debugf("azure appconfig error: fail to get sentinel key changes or sentinel key's value is unchanged: %s", err)
+			if !errors.Is(err, context.Canceled) {
```

Comment:
If the error is context.Canceled(), shouldn't the method return?

Reply:
Yes, agree. My mistake

Reply:
@ItalyPaleAle

Reply:
You're not wrong, but I do believe that we should short-circuit when we can. This way if someone adds more code in the future, they are less likely to accidentally introduce a bug.